testsuite/tests/libratest/smoke_test.rs

```diff
@@ -336,3 +340,62 @@ fn test_full_node() {
     let response = client_proxy.mint_coins(&["mint", "0", "1"], false);
     assert!(response.is_err());
 }
+
+#[test]
+fn test_external_transaction_signer() {
+    let (_swarm, mut client_proxy) = setup_swarm_and_client_proxy(1, 0);
+
+    // generate key pair
+    let mut seed: [u8; 32] = [0u8; 32];
+    seed[..4].copy_from_slice(&[1, 2, 3, 4]);
+    let mut rng: StdRng = SeedableRng::from_seed(seed);
+    let key_pair = KeyPair::<Ed25519PrivateKey, Ed25519PublicKey>::generate_for_testing(&mut rng);
+
+    // create transfer parameters
+    let sender_address = AccountAddress::from_public_key(&key_pair.public_key);
+    let receiver_address = client_proxy
+        .get_account_address_from_parameter(
+            "1bfb3b36384dabd29e38b4a0eafd9797b75141bb007cea7943f8a4714d3d784a",
+        )
+        .unwrap();
+    let amount = ClientProxy::convert_to_micro_libras("1").unwrap();
+
+    // mint to the sender address
+    client_proxy
+        .mint_coins(&["mintb", &format!("{}", sender_address), "10"], true)
+        .unwrap();
+
+    // prepare transfer transaction
+    let sequence_number = client_proxy
+        .get_sequence_number(&["sequence", &format!("{}", sender_address)])
+        .unwrap();
+
+    let unsigned_txn = client_proxy
+        .prepare_transfer_coins(
+            sender_address,
+            sequence_number,
+            receiver_address,
+            amount,
+            None, /* gas_unit_price */
+            None, /* max_gas_amount */
+        )
+        .unwrap();
+
+    // extract the hash to sign from the raw transaction
+    let raw_bytes = unsigned_txn.clone().into_proto_bytes().unwrap();
+    let txn_hashvalue = RawTransactionBytes(&raw_bytes).hash();
+
+    // sign the transaction with the private key
+    let signature = key_pair.private_key.sign_message(&txn_hashvalue);
+
+    // submit the transaction
+    let address_and_sequence = client_proxy
+        .submit_signed_transaction(unsigned_txn, key_pair.public_key, signature)
+        .unwrap();
+
+    assert_eq!(
+        address_and_sequence.account_address.to_string(),
+        format!("{}", sender_address)
+    );
+    assert_eq!(address_and_sequence.sequence_number, 0);
```

Comment:
Better add asserts for every property, amounts, gas etc to ensure we capture issues in upcoming refactorings.

Reply:
Added, asserted equality of all submitted transaction's fields with their respective input values.